### DISCUSSION TITLE
Can multiple instances of immich_server use the same immich_machine_learning container?

### DISCUSSION THREAD

**Discussion by imjuzcy** [immich-app/immich#14321] [Q&A] [answered]
I'm hosting two instances of Immich, that I do not want to merge/mix up. However, one of the instance is hosted on a Raspberry Pi, which is not ideal for the machine learning (not enough RAM as I'm also running other things on it). I'm currently using a `immich_machine_leaning` container on an external VPS for this instance of Immich. This VPS is commissioned just for this purpose.

Another instance of Immich is hosted on a more powerful system, alongside the other accompanying containers, including the `immich_machine_learning` it's using.

So the question is, can I make two `immich_server` instances use the same `immich_machine_learning` container that is hosting on the more powerful system, providing that they use the same models for smart search and facial recognition? Or will this make them mix up to some degree?

**Comment by bo0tzz** [comment, marked as answer]
Yes. They can also use different models, at the cost of extra RAM.

**Reply by Eric-YOUNGOUA** [reply]
I want the opposite, i have a very powerfull server so i want to know if  can accelerate the machine learning tasks by maybe having 3 or 4 immich machine learning container runnung alongside. I want the tasks to be done like instantly.

**Reply by bo0tzz** [reply]
Yes, but you need to handle loadbalancing between them yourself.
